Core/Src/sensors/encoder_speed.c

```diff
@@ -59,63 +60,61 @@ void encoder_speed_calc(void) {
+            // case the task was called by a new message arriving
             default:
-                // verifica se alguma roda esta a muito tempo sem receber interrupcao,
-                // caso sim a velocidade dessa roda eh zerada
-                reset_speed_single(&message, last_messages, min_count);
+                // verifies if any wheel is without an interruption for a long time,
+                // if yes that wheel speed is zeroed
+                reset_speed_single(&interrupt_message, last_interrupt_messages,
+                                   min_count);
 
-                // diferenca entre timestamp da mensagem atual e da anterior
-                d_tim_count = message.tim_count - last_messages[message.pin].tim_count;
+                // difference between current message and last message timestamp
+                d_tim_count = interrupt_message.tim_count
+                              - last_interrupt_messages[interrupt_message.pin].tim_count;
 
-                // caso d_tim_count calcule uma velocidade maior do que eh possivel ela
-                // sera descartada
+                // discards value if d_tim_count results in a speed greater than that
+                // configured as maximum
                 if (d_tim_count < max_count) {
                     continue;
                 }
 
                 speed = calculate_speed(d_tim_count, tim_freq, tim_presc);
-                WHEEL_ENCODER_SPEEDS_t encoder_speed =
-                    get_global_var_value(WHEEL_ENCODER_SPEEDS);
-                // seta velocidade especifica da roda recebida
-                encoder_speed.wheels[message.pin] = speed;
-                set_global_var(WHEEL_ENCODER_SPEEDS, &encoder_speed);
-                // guarda mensagem ate a proxima interacao
-                last_messages[message.pin] = message;
+                // saves the speed only of the wheel which speed was just calculated
+                speeds_message.wheels[interrupt_message.pin] = speed;
+
+                // store message to use in the next iteration
+                last_interrupt_messages[interrupt_message.pin] = interrupt_message;
                 break;
         }
+        osMessageQueuePutOverwrite(q_encoder_speeds_messageHandle, &speeds_message, 0);
     }
 }
 
-void reset_speed_all() {
-    WHEEL_ENCODER_SPEEDS_t encoder_speed;
+static void reset_speed_all() {
     for (uint8_t i = 0; i < WHEEL_ENCODERS_AVAILABLE; i++) {
-        encoder_speed.wheels[i] = 0;
+        speeds_message.wheels[i] = 0;
     }
-    set_global_var(WHEEL_ENCODER_SPEEDS, &encoder_speed);
 }
 
-void reset_speed_single(speed_message_t* message, speed_message_t* last_messages,
-                        uint32_t min_count) {
-    WHEEL_ENCODER_SPEEDS_t encoder_speed = get_global_var_value(WHEEL_ENCODER_SPEEDS);
+static void reset_speed_single(encoder_int_message_t* message,
+                               encoder_int_message_t* last_messages, uint32_t min_count) {
```

Comment:
```suggestion
static void reset_speed_single(const encoder_int_message_t* message,
                               const encoder_int_message_t* last_messages, uint32_t min_count) {
```
(vê se ele deixa, aparentemente não deve ser problema)